Create helperLib.py

WK-GiHu opened the pull request:
Helper class and def's used in modified osc_modif.py

WK-GiHu committed:
Create helperLib.py

Helper class and def's used in modified osc_modif.py

jocelynj commented on osc_modif/modules/helperLib.py:
```diff
@@ -0,0 +1,89 @@
+    """ Read file data to md5 buffer """
+    self.md5 = hashlib.md5()
+    self.sDescript = fpath
+    f = io.open(fpath); self.md5.update( f.read() );f.close()
```
Are you sure that this syntax works in python ?
In any case, I would prefer one "command" per line

WK-GiHu replied:
Yes, all code tested.
The semicolon ( ; ) allows multiple statements on the single line given that neither statement starts a new code block.

jocelynj commented on osc_modif/modules/helperLib.py:
```diff
@@ -0,0 +1,89 @@
+  if i >= 0: regex = name[0:i]+'\\'+name[i:len(name)]
+    # Make Star a re.any = .*
+  i = regex.find('*')
+  if i >= 0: regex = regex[0:i]+'.'+regex[i:len(regex)]
```
Wouldn't it better to directly pass a regexp to remove() instead of a string with ? and \* ?
I'm fine if you want to keep the current code, as it is more shell-like, but it adds a few lines of code.

WK-GiHu replied:
My intention was to allow clients (eg. unittest.Test())  to use simple arguments, shell-like is simple.
In python you would have to hand over: (**`"r'.*\.osc'"`**), including double and single quotes!
Regex ist powerfull but seldom got it matching on the first try.

jocelynj commented:
Changes looks fine, I can merge them as soon as you answer my review questions.

What is the purpose of the shakeUp() function ?

WK-GiHu commented:
What is the purpose of the shakeUp() function ?

Proposed for helperLib.unittest.
I used it to verify `class hash_file():` will work as expected.

WK-GiHu committed:
Update helperLib.py with unittest

Unittest `remove(...)` shows some mistakes.
Corrected, passed all tests.
Also osc_modif.py Version 102 tested.

WK-GiHu committed:
Update helperLib.py

jocelynj commented:
Thanks for the changes. I've pushed them to master, with some minor modifications.

About the regexp changes: is there a reason to not replace all "*" pay ".+" ?
